tests/conftest.py

```diff
@@ -153,6 +154,20 @@ def requires(self):
             return self.dep_strs
 
 
+class FakeInstalledDistribution:
```

Comment:
Could you explain for what reasons exactly you derived this from the original `FakeInstalledDistribution`?

Reply:
I added some more comments, but I am not sure I understood exactly what you meant.

If you want, I can clean up this pattern by using an adapter class in the compatibility layer, which will have the attributes name, version, key, requires and direct_url. Then the typing will be much cleaner in both the code and the tests.

Reply:
That would be great, thanks.

I meant to ask why you created inner and outer `FakeInstalledDistribution`.

Reply:
Alright, I made an adapter class. I think the new pattern is much cleaner, but let me know if you would like me to revert to the old one.

Reply:
Tests are failing for Python 3.7 because `importlib-metadata` is missing from the test environment. Should I add it to setup.cfg. Not sure how you would like me to proceed here.

Reply:
It can be added with an env marker so it's installed conditionally.

Reply:
Alright. I did something. We will know for sure when the checks run.